frontend/src/components/Login.jsx

```diff
@@ -1,89 +1,255 @@
-import "../css/Login.css";
-import logo from '../assets/logo.png';
-import { useAuth } from './AuthContext';
-import { useNavigate } from 'react-router-dom';
+//Logic packages
+import { useAuth } from "./AuthContext";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import queryString from "query-string";
+
+//React components
+import {
+  Container,
+  Row,
+  Col,
+  Button,
+  Form,
+  Image,
+  InputGroup,
+} from "react-bootstrap";
+import { ArrowLeft, Eye, EyeSlash } from "react-bootstrap-icons";
+
+//Styles
+import "../css/common.css";
+
+//Assets
+import logo from "../assets/logo.png";
 
 function Login() {
-    const [username, setUsername] = useState('');
-    const [password, setPassword] = useState('');
-    const [passwordError, setPasswordError] = useState('');
-    const [showPassword, setShowPassword] = useState(false);
-    const { token, setToken } = useAuth();
-    const navigate = useNavigate();
-
-    const handleSubmit = async (e) => {
-        e.preventDefault();
-
-        const apiUrl = 'http://127.0.0.1:8000/user/token';
-        const requestBody = queryString.stringify({
-            grant_type: '',
-            username,
-            password,
-            scope: '',
-            client_id: '',
-            client_secret: ''
-        });
-
-        try {
-            const response = await fetch(apiUrl, {
-                method: 'POST',
-                headers: {
-                    'Content-Type': 'application/x-www-form-urlencoded',
-                    'accept': 'application/json',
-                },
-                body: requestBody
-            });
-
-            if (response.ok) {
-                const data = await response.json();
-                console.log('Token obtenido:', data.access_token);
-                setToken(data.access_token)
-                navigate("/userfeed");
-                
-            } else {
-                const errorData = await response.json();
-                setPasswordError(errorData.detail || "Error al iniciar sesión.");
-            }
-        } catch (error) {
-            setPasswordError("Hubo un problema al conectar con el servidor.");
+  const [identifier, setIdentifier] = useState("");
+  const [password, setPassword] = useState("");
+  const { token, setToken } = useAuth();
+
+  const navigate = useNavigate();
+
+  const [showPassword, setShowPassword] = useState(false);
+
+  const [identifierFilled, setIdentifierFilled] = useState(false);
+  const [passwordFilled, setPasswordFilled] = useState(false);
+
+  const [identifierValidated, setIdentifierValidated] = useState(false);
+  const [passwordValidated, setPasswordValidated] = useState(false);
+
+  const [loginError, setLoginError] = useState(false);
+  const [loginErrorMessage, setLoginErrorMessage] = useState("");
+
+  /* Requests */
+
+  const loginRequest = async (identifier, password) => {
+    const api_url = "http://0.0.0.0:8000/user/token";
```

Comment:
For security reasons, we shouldn't hardcode credentials or URLs directly in the code. Consider using environment variables or some other configuration mechanism.

Reply:
I commented on this in the US7 PR